packages/frontend-app-api/src/wiring/createPluginInfoAttacher.ts

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConfigApi } from '@backstage/core-plugin-api';
+import {
+  FrontendFeature,
+  FrontendPluginInfo,
+} from '@backstage/frontend-plugin-api';
+import { OpaqueFrontendPlugin } from '@internal/frontend';
+import { JsonObject, JsonValue } from '@backstage/types';
+import once from 'lodash/once';
+// Avoid full dependency on catalog-model
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import {
+  parseEntityRef,
+  stringifyEntityRef,
+} from '../../../catalog-model/src/entity/ref';
+
+/**
+ * A function that resolves plugin info from a plugin manifest and package.json.
+ *
+ * @public
+ */
+export type FrontendPluginInfoResolver = (ctx: {
+  packageJson?(): Promise<JsonObject>;
+  manifest?(): Promise<JsonObject>;
+  defaultResolver(overrides?: {
+    packageJson?: JsonObject;
+    manifest?: JsonObject;
+  }): Promise<{ info: FrontendPluginInfo }>;
+}) => Promise<{ info: FrontendPluginInfo }>;
+
+export function createPluginInfoAttacher(
+  config: ConfigApi,
+  infoResolver: FrontendPluginInfoResolver = ctx => ctx.defaultResolver(),
+): (feature: FrontendFeature) => FrontendFeature {
+  const applyInfoOverrides = createPluginInfoOverrider(config);
+
+  return (feature: FrontendFeature) => {
+    if (!OpaqueFrontendPlugin.isType(feature)) {
+      return feature;
+    }
+
+    const plugin = OpaqueFrontendPlugin.toInternal(feature);
+
+    return {
+      ...plugin,
+      info: once(async () => {
```

Comment:
Is there something to be said about error handling? These are going to be pretty "late" runtime errors if something goes wrong right? How will those errors generally be noticed by users in the end? Can for example a slight syntax error in a custom field in a catalog-info make the entire interface be unusable? Or will it generally be more localized problems for example inside the support button and similar places that use the new hook for that one single plugin page?

Reply:
Yep I aimed to push errors as late as possible everywhere, leaving it up to projects to validate manifests if needed or just generally being very permissive with the input. The intention is to have errors pop up pretty deep related to support buttons or error boundaries, but not completely break those either.